### PR TITLE
add a patched libcurl v7 build for use with Julia 1.6

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: L/LibCURL/LibCURL@7/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libcurl", :libcurl),
+    ExecutableProduct("curl", :curl),
```
[the remark this answers is not in the document]
Comment (KristofferC) [reply L/LibCURL/LibCURL@7/build_tarballs.jl:62]: I took this from the previous v7 script at https://github.com/JuliaPackaging/Yggdrasil/blob/4c697e5d37d6c6eb703ed6b60596f2bda3298518/L/LibCURL/build_tarballs.jl. But I can remove it.

[the remark this answers is not in the document]
Comment (giordano) [reply L/LibCURL/LibCURL@7/build_tarballs.jl]: That probably predates #2720 and  #4472

**Comment by IanButterworth** [comment]
Perhaps drop Apple aarch64? Not a supported platform on 1.6?

**Comment by giordano** [comment]
#2439

**Pull request merged**